python/pyspark/sql/functions.py

```diff
@@ -13041,6 +13041,120 @@ def json_object_keys(col: "ColumnOrName") -> Column:
     return _invoke_function_over_columns("json_object_keys", col)
 
 
+@_try_remote_functions
+def from_xml(
+    col: "ColumnOrName",
+    schema: Union[StructType, Column, str],
+    options: Optional[Dict[str, str]] = None,
+) -> Column:
+    """
+    Parses a column containing a XML string to a row with
+    the specified schema. Returns `null`, in the case of an unparseable string.
+
+    .. versionadded:: 4.0.0
+
+    Parameters
+    ----------
+    col : :class:`~pyspark.sql.Column` or str
+        a column or column name in XML format
+    schema : :class:`StructType` or str
+        a StructType or Python string literal with a DDL-formatted string
+        to use when parsing the Xml column
+    options : dict, optional
+        options to control parsing. accepts the same options as the Xml datasource.
+        See `Data Source Option <https://spark.apache.org/docs/latest/sql-data-sources-xml.html#data-source-option>`_
+        for the version you use.
+
+        .. # noqa
+
+    Returns
+    -------
+    :class:`~pyspark.sql.Column`
+        a new column of complex type from given XML object.
+
+    Examples
```

Comment:
I think we should improve the examples here but let's do that separately. cc @allisonwang-db and @zhengruifeng FYI

Reply:
Documentation is extremely important for a better user-experience. @sandip-db could you please create a ticket under https://issues.apache.org/jira/browse/SPARK-44728.

Reply:
Yeah, we can do it in separate PR, but let's simply put some newline to readability for now.

e.g.

```python
    >>> from pyspark.sql.types import *
    >>> from pyspark.sql.functions import from_xml, schema_of_xml, lit

    StructType input with simple IntegerType.

    >>> data = [(1, '''<p><a>1</a></p>''')]
    >>> df = spark.createDataFrame(data, ("key", "value"))
    >>> schema = StructType([StructField("a", IntegerType())])
    >>> df.select(from_xml(df.value, schema).alias("xml")).collect()
    [Row(xml=Row(a=1))]

    String input.

    >>> df.select(from_xml(df.value, "a INT").alias("xml")).collect()
    [Row(xml=Row(a=1))]

    StructType input with complex ArrayType.

    >>> data = [(1, '<p><a>1</a><a>2</a></p>')]
    >>> df = spark.createDataFrame(data, ("key", "value"))
    >>> schema = StructType([StructField("a", ArrayType(IntegerType()))])
    >>> df.select(from_xml(df.value, schema).alias("xml")).collect()
    [Row(xml=Row(a=[1, 2]))]

    Column input generated by schema_of_xml.

    >>> schema = schema_of_xml(lit(data[0][1]))
    >>> df.select(from_xml(df.value, schema).alias("xml")).collect()
    [Row(xml=Row(a=[1, 2]))]
```

Reply:
> could you please create a ticket
@allisonwang-db 
Sure, created a ticket here: https://issues.apache.org/jira/browse/SPARK-45186